redis/retry.py

```diff
@@ -1,17 +1,27 @@
 import socket
 from time import sleep
+from typing import TYPE_CHECKING, Any, Iterable, TypeVar, Callable
 
 from redis.exceptions import ConnectionError, TimeoutError
 
+T = TypeVar("T")
+
+if TYPE_CHECKING:
+    from redis.backoff import AbstractBackoff
+
 
 class Retry:
     """Retry a specific number of times after a failure"""
 
     def __init__(
         self,
-        backoff,
-        retries,
-        supported_errors=(ConnectionError, TimeoutError, socket.timeout),
+        backoff: AbstractBackoff,
```

Comment:
You're right, went ahead and quoted it: https://github.com/redis/redis-py/pull/3250/commits/16603366a4cf113fbb41e510cdafd69d6cb8944e

It would be nicer to quote, just to avoid an unnecessary import which would slow down module init time.